### PR TITLE
Feat(eos_cli_config_gen): Add support for LSP and CSNP timers under router_isis

### LINKED ISSUE #4287: Additional support to modify ISIS timers
### Enhancement summary

For Service Provider deployments we commonly need to tweak ISIS underlay timers for faster convergence. In particular the SPF-interval, LSP generation timer and sometimes the LSP out-delay timer is updated:

https://www.arista.com/en/support/toi/eos-4-15-2f/13692-isis-spf-timer

Starting  4.31.2F:
```
spf-interval <max-wait> seconds | milliseconds  <initial-wait> milliseconds <hold-time> milliseconds
```
LSP generation command:
```
ps-rtp1-leaf1(config-router-isis)#timers lsp generation ?
  <1-300>  Maximum interval (in seconds) between generating two LSPs
ps-rtp1-leaf1(config-router-isis)#timers lsp generation 5 ?
  <1-300000>  Initial wait time (in milliseconds) before generating LSPs
ps-rtp1-leaf1(config-router-isis)#timers lsp generation 5 200 ?
  <1-300000>  Wait time (in milliseconds) between generating the first and second LSPs
  <cr>
```
LSP out-delay:
```
ps-rtp1-leaf1(config-router-isis)#timers lsp out-delay ?
  <1-65000>  Transmit delay (in milliseconds) for link state packets
```

There are other ISIS timers in case you want to consider to include as part of this change (these are not urgent for us):
https://www.arista.com/en/support/toi/eos-4-21-3f/14162-isis-lsp-refresh-interval
https://www.arista.com/en/support/toi/eos-4-27-0f/14859-is-is-csnp-generation-interval

### Which component of AVD is impacted

eos_cli_config_gen

### Use case example

Service Provider deployments, modify ISIS default timers.

### Describe the solution you would like

Include the previous commands for ISIS:
https://avd.arista.com/4.10/roles/eos_cli_config_gen/docs/input-variables.html#router-isis

### Describe alternatives you have considered

_No response_

### Additional context

_No response_

### Contributing Guide

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by Shivani-gslab**
## Change Summary

Add support for LSP and CSNP timers under router_isis

## Related Issue(s)

Fixes #4287 

## Component(s) name

`arista.avd.eos_cli_config_gen`

## Proposed changes
```
router_isis:
    timers:
        # Link State Packet timers
        lsp:
          generation:

            # Maximum interval (in seconds) between generating two LSPs.
            interval: <int; 1-300; required>

            # Initial wait time (in milliseconds) before generating LSPs.
            initial_wait_time: <int; 1-300000>

            # Wait time (in milliseconds) between generating the first and second LSPs.
            wait_time: <int; 1-300000>

          # Transmit delay (in milliseconds) for link state packets.
          out_delay: <int; 1-65000>

          # Interval (in seconds) between two LSP refreshes.
          refresh_interval: <int; 30-65535>

          # Minimum remaining lifetime for LSPs (in seconds).
          min_remaining_lifetime: <int; 60-65535>
        csnp_generation:

          # Transmit frequency (in seconds) for CSN packets.
          interval: <int; 1-300>

          # Disable periodic CSN packets.
          p2p_disabled: <bool>
```

## How to test
Molecule test and check EOS CLI.

### User Checklist

- N/A

### Repository Checklist

- [x] My code has been rebased from devel before I start
- [x] I have read the [**CONTRIBUTING**](https://avd.sh/en/latest/docs/contribution/overview.html) document.
- [x] My change requires a change to the documentation and documentation have been updated accordingly.
- [x] I have updated [molecule CI](https://github.com/aristanetworks/avd/tree/devel/ansible_collections/arista/avd/molecule) testing accordingly. (check the box if not applicable)

### CODE REVIEW

File: python-avd/pyavd/_eos_cli_config_gen/schema/schema_fragments/router_isis.schema.yml [hunk 1]
```diff
@@ -44,6 +44,69 @@ keys:
...
+              interval:
+                type: int
+                convert_types:
+                  - str
+                min: 1
+                max: 300
+                description: Transmit frequency (in seconds) for CSN packets.
+              p2p_disabled:
+                type: bool
+                description: Disable periodic CSN packets.
```
Comment (laxmikantchintakindi) [inline python-avd/pyavd/_eos_cli_config_gen/schema/schema_fragments/router_isis.schema.yml]:
```suggestion
                description: Disable periodic CSN packets for p2p links.
```

File: python-avd/pyavd/_eos_cli_config_gen/j2templates/eos/router-isis.j2 [hunk 2]
```diff
@@ -91,6 +91,30 @@ router isis {{ router_isis.instance }}
...
+   timers lsp refresh {{ router_isis.timers.lsp.refresh_interval }}
+{%     endif %}
+{%     if router_isis.timers.lsp.generation.interval is arista.avd.defined %}
+{%         set timers_lsp_generation =  "timers lsp generation " ~ router_isis.timers.lsp.generation.interval %}
+{%         if router_isis.timers.lsp.generation.initial_wait_time is arista.avd.defined %}
+{%             set timers_lsp_generation = timers_lsp_generation ~ " " ~ router_isis.timers.lsp.generation.initial_wait_time %}
+{%             if router_isis.timers.lsp.generation.wait_time is arista.avd.defined %}
+{%                 set timers_lsp_generation = timers_lsp_generation ~ " " ~ router_isis.timers.lsp.generation.wait_time %}
+{%             endif %}
+{%         endif %}
```
Comment (laxmikantchintakindi) [inline python-avd/pyavd/_eos_cli_config_gen/j2templates/eos/router-isis.j2:113]:
```suggestion
{%         endif %}
   {{ timers_lsp_generation }}
```

**Review by laxmikantchintakindi — APPROVED** [review]

File: python-avd/pyavd/_eos_cli_config_gen/schema/schema_fragments/router_isis.schema.yml [hunk 3]
```diff
@@ -44,6 +44,69 @@ keys:
                   - str
                 default: 10000
                 description: Delay in milliseconds.
+          lsp:
+            description: Link State Packet timers
```
Comment (MaheshGSLAB) [inline python-avd/pyavd/_eos_cli_config_gen/schema/schema_fragments/router_isis.schema.yml]:
```suggestion
            description: Link State Packet timers.
```

File: ansible_collections/arista/avd/molecule/eos_cli_config_gen/intended/configs/router-isis.cfg [hunk 4]
```diff
@@ -108,6 +108,12 @@ router isis EVPN_UNDERLAY
    set-overload-bit on-startup wait-for-bgp
    advertise passive-only
    spf-interval 250
+   timers csnp generation interval 10 seconds
```
Comment (MaheshGSLAB) [inline ansible_collections/arista/avd/molecule/eos_cli_config_gen/intended/configs/router-isis.cfg:111]: can you improve coverage for your changes?

File: python-avd/pyavd/_eos_cli_config_gen/schema/schema_fragments/router_isis.schema.yml (hunk 3 above)
Comment (Shivani-gslab) [reply python-avd/pyavd/_eos_cli_config_gen/schema/schema_fragments/router_isis.schema.yml]: Fixed.

File: ansible_collections/arista/avd/molecule/eos_cli_config_gen/intended/configs/router-isis.cfg (hunk 4 above)
Comment (Shivani-gslab) [reply ansible_collections/arista/avd/molecule/eos_cli_config_gen/intended/configs/router-isis.cfg:111]: Fixed

**Review by MaheshGSLAB — APPROVED** [review]

**Review by ClausHolbechArista — DISMISSED** [review]
Blocking until 5.0 is out

File: python-avd/pyavd/_eos_cli_config_gen/schema/schema_fragments/router_isis.schema.yml [hunk 5]
```diff
@@ -44,6 +44,69 @@ keys:
...
+              interval:
+                type: int
+                convert_types:
+                  - str
+                min: 1
+                max: 300
+                description: Transmit frequency (in seconds) for CSN packets.
+              p2p_disabled:
+                type: bool
+                description: Disable periodic CSN packets for P2P links.
```
Comment (gmuloc) [inline python-avd/pyavd/_eos_cli_config_gen/schema/schema_fragments/router_isis.schema.yml]: I discussed this with other maintainers on a calll ; from a consistency point of view with `lsp` implemented in the same PR let's use a "more than necessary" nested dictionary here.

```suggestion
          csnp:
            type: dict
            keys:
              generation:
                type: dict
                keys:
                  interval:
                    type: int
                    convert_types:
                      - str
                    min: 1
                    max: 300
                    description: Transmit frequency (in seconds) for CSN packets.
                  p2p_disabled:
                    type: bool
                    description: Disable periodic CSN packets for P2P links.
```

**Review by gmuloc — CHANGES_REQUESTED** [review]

File: ansible_collections/arista/avd/molecule/eos_cli_config_gen/documentation/devices/router-isis.md [hunk 6]
```diff
@@ -282,6 +282,14 @@ interface Vlan4094
...
 | Log Adjacency Changes | True |
 | Local Convergence Delay (ms) | 15000 |
+| CSN Packet Transmission Interval | 10 seconds |
+| CSN Packet P2P Links Disabled | True |
+| LSP Generation Maximum Interval | 30 seconds |
+| LSP Generation Initial Wait-time | 40 milliseconds |
+| LSP Generation Wait-time | 50 milliseconds |
+| LSP Out-delay | 20 milliseconds |
+| LSP Refresh Interval | 56 seconds |
+| LSP Minimum Remaining Lifetime | 78 seconds |
```
Comment (gmuloc) [inline ansible_collections/arista/avd/molecule/eos_cli_config_gen/documentation/devices/router-isis.md:300]: let's make a table for timers only so it is clear

**Review by gmuloc — CHANGES_REQUESTED** [review]

File: ansible_collections/arista/avd/molecule/eos_cli_config_gen/documentation/devices/router-isis.md (hunk 6 above)
Comment (Shivani-gslab) [reply ansible_collections/arista/avd/molecule/eos_cli_config_gen/documentation/devices/router-isis.md:300]: added

File: ansible_collections/arista/avd/molecule/eos_config_deploy_cvp/documentation/devices/DC1-BL1A.md [hunk 7]
```diff
@@ -690,9 +690,9 @@ ASN Notation: asplain
 
 ##### EVPN Peer Groups
 
-| Peer Group | Activate | Encapsulation |
-| ---------- | -------- | ------------- |
-| EVPN-OVERLAY-PEERS | True | default |
+| Peer Group | Activate | Route-map In | Route-map Out | Encapsulation |
+| ---------- | -------- | ------------ | ------------- | ------------- |
+| EVPN-OVERLAY-PEERS | True |  - | - | default |
```
Comment (Shivani-gslab) [inline ansible_collections/arista/avd/molecule/eos_config_deploy_cvp/documentation/devices/DC1-BL1A.md:696]: getting these unexpected diffs in multiple files, this needs to be fixed in `devel`.
is it okay to keep it in this PR or shall I do a separate PR for this?

Comment (gmuloc) [reply ansible_collections/arista/avd/molecule/eos_config_deploy_cvp/documentation/devices/DC1-BL1A.md:696]: let's keep them here that was missed in a previous PR becayse of CI not running these. Will adjust CI later to not miss it again.

File: python-avd/pyavd/_eos_cli_config_gen/j2templates/documentation/router-isis.j2 [hunk 8]
```diff
@@ -30,9 +30,6 @@
 {%     if router_isis.mpls_ldp_sync_default is arista.avd.defined(true) %}
 | MPLS LDP Sync Default | {{ router_isis.mpls_ldp_sync_default }} |
 {%     endif %}
-{%     if router_isis.timers.local_convergence.protected_prefixes is arista.avd.defined(true) %}
-| Local Convergence Delay (ms) | {{ router_isis.timers.local_convergence.delay | arista.avd.default(10000) }} |
```
Comment (gmuloc) [inline python-avd/pyavd/_eos_cli_config_gen/j2templates/documentation/router-isis.j2:34]: Maybe let's move `ms` after the value to look like the other

Comment (Shivani-gslab) [reply python-avd/pyavd/_eos_cli_config_gen/j2templates/documentation/router-isis.j2:34]: Fixed

**Review by gmuloc — APPROVED** [review]
